lib/class-wp-rest-widget-updater-controller.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * Start: Include for phase 2
+ * Widget Updater REST API: WP_REST_Widget_Updater_Controller class
+ *
+ * @package gutenberg
+ * @since 5.2.0
+ */
+
+/**
+ * Controller which provides REST endpoint for updating a widget.
+ *
+ * @since 5.2.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Widget_Updater_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructs the controller.
+	 *
+	 * @access public
+	 */
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'widgets';
+	}
+
+	/**
+	 * Registers the necessary REST API route.
+	 *
+	 * @access public
+	 */
+	public function register_routes() {
+			register_rest_route(
+				$this->namespace,
+				// Regex representing a PHP class extracted from http://php.net/manual/en/language.oop5.basic.php.
+				'/' . $this->rest_base . '/(?P<identifier>[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)/',
+				array(
+					'args' => array(
+						'identifier' => array(
```

Comment:
Hi @noisysocks,
I used identifier because this field is not the widget id, it may be the widget id for call back widgets after https://github.com/WordPress/gutenberg/pull/14395 is landed, or it may be the class name for class widgets.
So in order to differentiate from the widget id I used identifier.